lease/lessor_bench_test.go

```diff
@@ -15,112 +15,237 @@
 package lease
 
 import (
+	"math/rand"
 	"os"
 	"testing"
+	"time"
 
 	"go.etcd.io/etcd/v3/mvcc/backend"
 	"go.uber.org/zap"
 )
 
-func BenchmarkLessorFindExpired1(b *testing.B)       { benchmarkLessorFindExpired(1, b) }
-func BenchmarkLessorFindExpired10(b *testing.B)      { benchmarkLessorFindExpired(10, b) }
-func BenchmarkLessorFindExpired100(b *testing.B)     { benchmarkLessorFindExpired(100, b) }
-func BenchmarkLessorFindExpired1000(b *testing.B)    { benchmarkLessorFindExpired(1000, b) }
-func BenchmarkLessorFindExpired10000(b *testing.B)   { benchmarkLessorFindExpired(10000, b) }
-func BenchmarkLessorFindExpired100000(b *testing.B)  { benchmarkLessorFindExpired(100000, b) }
-func BenchmarkLessorFindExpired1000000(b *testing.B) { benchmarkLessorFindExpired(1000000, b) }
-
-func BenchmarkLessorGrant1(b *testing.B)       { benchmarkLessorGrant(1, b) }
-func BenchmarkLessorGrant10(b *testing.B)      { benchmarkLessorGrant(10, b) }
-func BenchmarkLessorGrant100(b *testing.B)     { benchmarkLessorGrant(100, b) }
-func BenchmarkLessorGrant1000(b *testing.B)    { benchmarkLessorGrant(1000, b) }
-func BenchmarkLessorGrant10000(b *testing.B)   { benchmarkLessorGrant(10000, b) }
-func BenchmarkLessorGrant100000(b *testing.B)  { benchmarkLessorGrant(100000, b) }
-func BenchmarkLessorGrant1000000(b *testing.B) { benchmarkLessorGrant(1000000, b) }
-
-func BenchmarkLessorRenew1(b *testing.B)       { benchmarkLessorRenew(1, b) }
-func BenchmarkLessorRenew10(b *testing.B)      { benchmarkLessorRenew(10, b) }
-func BenchmarkLessorRenew100(b *testing.B)     { benchmarkLessorRenew(100, b) }
-func BenchmarkLessorRenew1000(b *testing.B)    { benchmarkLessorRenew(1000, b) }
-func BenchmarkLessorRenew10000(b *testing.B)   { benchmarkLessorRenew(10000, b) }
-func BenchmarkLessorRenew100000(b *testing.B)  { benchmarkLessorRenew(100000, b) }
-func BenchmarkLessorRenew1000000(b *testing.B) { benchmarkLessorRenew(1000000, b) }
-
-func BenchmarkLessorRevoke1(b *testing.B)       { benchmarkLessorRevoke(1, b) }
-func BenchmarkLessorRevoke10(b *testing.B)      { benchmarkLessorRevoke(10, b) }
-func BenchmarkLessorRevoke100(b *testing.B)     { benchmarkLessorRevoke(100, b) }
-func BenchmarkLessorRevoke1000(b *testing.B)    { benchmarkLessorRevoke(1000, b) }
-func BenchmarkLessorRevoke10000(b *testing.B)   { benchmarkLessorRevoke(10000, b) }
-func BenchmarkLessorRevoke100000(b *testing.B)  { benchmarkLessorRevoke(100000, b) }
-func BenchmarkLessorRevoke1000000(b *testing.B) { benchmarkLessorRevoke(1000000, b) }
+// NOTE: run with --benchtime Nx, please
```

Comment:
`Ns` working, however the actual use time may exceeding our exceptions because this implementation.
```
lease git:(refactor-lease-bench-test) ✗ ~/go/src/github.com/golang/go/bin/go test --bench BenchmarkLessorRevoke100000 -benchtime 1s
goos: darwin
goarch: amd64
pkg: go.etcd.io/etcd/v3/lease
BenchmarkLessorRevoke100000-4            8995898               133 ns/op
PASS
ok      go.etcd.io/etcd/v3/lease        83.956s
```
With `benchtime Ns`, golang estimate size of benchmark through the last time consuming.
We spend a lot of time to construct fixture and let golang ignore it.